ui/components/app/wallet-overview/eth-overview.js

```diff
@@ -120,9 +120,9 @@ const EthOverview = ({ className, showAddress }) => {
   }
 
   const isSwapsChain = useSelector(getIsSwapsChain);
-  const isSigningEnabled = useSelector((state) =>
-    checkIfMethodIsEnabled(state, EthMethod.SignTransaction),
-  );
+  const isSigningEnabled =
+    account.methods.includes(EthMethod.SignTransaction) &&
```

Comment:
This should be an or condition

Reply:
Good catch! Thank you [cb9c97e](https://github.com/MetaMask/metamask-extension/pull/23178/commits/cb9c97ed27d3cb4c9aa640630afc08d1ea9474ad)